plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1090,6 +1091,34 @@ public void testArithmeticPredicatePushdown()
         // TODO add coverage for other arithmetic pushdowns https://github.com/trinodb/trino/issues/14808
     }
 
+    @Test
+    public void testComplexExpressionOnApproximateNumericPredicatePushdown()
+    {
+        if (!hasBehavior(SUPPORTS_PREDICATE_EXPRESSION_PUSHDOWN)) {
+            assertThat(query("SELECT orderkey FROM orders WHERE totalprice > 10 OR shippriority = 0"))
+                    .isNotFullyPushedDown(FilterNode.class);
+            return;
+        }
+        assertThat(query("SELECT orderkey FROM orders WHERE totalprice > 10 OR shippriority = 0")).isFullyPushedDown();
+
+        try (TestTable testTable = new TestTable(
+                getQueryRunner()::execute,
+                "test_complex_expression_pushdown_",
+                "AS SELECT 1 as a_int, REAL '3.14' as a_real, DOUBLE '1.9891e30' as a_double")) {
```

Comment:
add a test with NaN stored in the remote database (for systems that do support storing NaN) 
and queries `col < 1`, `col > 1`, `col > -inf`, `col < +inf`.

Reply:
I'll address it as a follow-up PR